pkg/cmd/cli/config/smart_merge.go

```diff
@@ -13,13 +14,75 @@ import (
 	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 
+	osclient "github.com/openshift/origin/pkg/client"
 	"github.com/openshift/origin/pkg/cmd/flagtypes"
 )
 
 var invalidSafeStringSep = regexp.MustCompile(`[.:/]`)
 
+func getClusterNickname(clientCfg *client.Config) (string, error) {
```

Comment:
should we eventually stop scanning all clusters for `Host` matches in login and just look for the cluster stanza with this name?